compiler/rustc_metadata/src/locator.rs

```diff
@@ -547,6 +545,20 @@ impl<'a> CrateLocator<'a> {
                             continue;
                         }
                     }
+                    Err(MetadataError::VersionMismatch(found_version)) => {
+                        // The file was present and created by the same compiler version, but we
+                        // couldn't load it for some reason.  Give a hard error instead of silently
+                        // ignoring it, but only if we would have given an error anyway.
```

Comment:
this comment doesn't make sense with the code below it

Reply:
I'm not sure where it came from. Maybe it was correct in an old draft? I finished a draft from September last year just now.